crates/engine/util/src/reorg.rs

```diff
@@ -298,7 +297,7 @@ where
         }
 
         let tx_recovered =
-            tx.try_clone_into_recovered().map_err(|_| ProviderError::SenderRecoveryError)?;
+            tx.try_into_recovered().map_err(|_| ProviderError::SenderRecoveryError)?;
```

Comment:
I believe this should also work here now
```suggestion
            tx.try_into_recovered()?;
```

Reply:
This seems not to work because `try_into_recovered()` return the tx (Self) for the error. And no error type that could be handled.

```
   --> crates/engine/util/src/reorg.rs:299:36
    |
243 | ) -> RethResult<SealedBlock<BlockTy<Evm::Primitives>>>
    |      ------------------------------------------------- expected `RethError` because of this
...
299 |             tx.try_into_recovered()?;
    |                --------------------^ the trait `From<<<Evm as ConfigureEvm>::Primitives as NodePrimitives>::SignedTx>` is not implemented for `RethError`
    |                |
    |                this can't be annotated with `?` because it has type `Result<_, <<Evm as ConfigureEvm>::Primitives as NodePrimitives>::SignedTx>`
```

Reply:
ah I see, let's keep this, need to figure out to address